### ISSUE TITLE
Web installation is failing (ci-build 585)

### ISSUE THREAD

**Issue by rallek** [zikula/core#2462]
I have unpacked on comand line the ci-build tar.gz download on my shared hosting. Opening the website gives an error:

Fatal error: Uncaught exception 'RuntimeException' with message 'Unable to create the cache directory (/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/cache/prod) ' in /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:609 Stack trace: #0 /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(533): Symfony\Component\HttpKernel\Kernel->buildContainer() #1 /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(306): Symfony\Component\HttpKernel\Kernel->initializeContainer() #2 /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(128): Zikula\Bundle\CoreBundle\HttpKernel\ZikulaKernel->initializeContainer() #3 /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(85): Symfony\Component\HttpKernel\Kernel->boot() in /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php on line 609

I can create a new folder via ftp without any problem. The owner is the same than all other files.

**Comment by rallek** [comment]
Additional information: Install via cli is creating the folders but permissions of cache and logs are 755. I have to change to 777. 

But than I run into next trouble:


Warning: file_get_contents(/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/../app/config/parameters.yml): failed to open stream: Permission denied in /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/bootstrap.php on line 22

Warning: array_merge(): Argument #1 is not an array in /www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/bootstrap.php on line 24
Whoops, looks like something went wrong.
1/1 InvalidArgumentException in FileLocator.php line 44: The file "/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/config_.yml" does not exist.

    in FileLocator.php line 44
    at FileLocator->locate('/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/config_.yml', null, true) in FileLocator.php line 54
    at FileLocator->locate('/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/config_.yml') in YamlFileLoader.php line 41
    at YamlFileLoader->load('/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/config_.yml', null) in DelegatingLoader.php line 45
    at DelegatingLoader->load('/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/config_.yml') in ZikulaKernel.php line 54
    at ZikulaKernel->registerContainerConfiguration(object(DelegatingLoader)) in Kernel.php line 620
    at Kernel->buildContainer() in Kernel.php line 533
    at Kernel->initializeContainer() in ZikulaKernel.php line 306
    at ZikulaKernel->initializeContainer() in Kernel.php line 128
    at Kernel->boot() in ZikulaKernel.php line 85
    at ZikulaKernel->boot() in bootstrap.php line 34
    at include('/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/lib/bootstrap.php') in index.php line 16

**Comment by rallek** [comment]
PHP is 5.5.

which other info is needed?

**Comment by craigh** [comment]
It seems you repeatedly run into file permissions/ownership problems on your host. These remind me of similar problems you have reported in the past. Your host is apparently not very friendly to these types of installations. I suggest you ask your host to help you install and see if the problems persist.

**Comment by rallek** [comment]
NO! I totally disagree with this behavior! All-Inkl is a very serious provider in Germany. All my sites are running there. All of you are testing in local environment or on own servers. It seem I am the only one testing on a sheared hosting.

We have to figure out what the reason is. the build 568 was able to instal. Now something is changing. Install now failed. Just to say: "The host should fix it" will reduce the community to some nerds! This was never the behavior of our community!

The new installer is not that robust what we need!

**Comment by cmfcmf** [comment]
Could you please check the permissions of the `app`, `app/config` and `app/config/parameters.yml` files and folders?

**Comment by rallek** [comment]
``all/config`` I already changed to 777. ``app/config/parameters.yml`` is 400.

changing parameters.yml to 777 brings the installation to an end.

Why all these permissions are always wrong in my case? Which change the host should do to make it nice?

**Comment by cmfcmf** [comment]
It might be an umask problem. https://github.com/zikula/core/pull/2434

**Comment by cmfcmf** [comment]
What is the result of `echo umask();`?

**Comment by rallek** [comment]
on the comand line?
````
$ echo umask();
bash: syntax error near unexpected token `('
````

**Comment by shefik** [comment]
@rallek I looked at your phpinfo file. I see that you have mod_suexec installed in your Apache. Check the setting there, to make sure it's correctly configured.

**Comment by cmfcmf** [comment]
No, I meant in PHP, sorry. (create a test.php file on the webserver and then open the url in the browser).

**Comment by rallek** [comment]
@shefik I am on shered hosting and con not change Apache settings.
@cmfcmf I will do

**Comment by rallek** [comment]
@cmfcmf A white page with ``18`` on it

**Comment by shefik** [comment]
@rallek I don't think you are allowed to use 777 on your server configuration, due to security restrictions with suExec. Try settings directories to 755 and files to 644.

**Comment by shefik** [comment]
For cross-reference, here is the documentation from WordPress, that address the similar issue with suExec:

https://codex.wordpress.org/Changing_File_Permissions#Shared_Hosting_with_suexec

`The above may not apply to shared hosting systems that use the "suexec" approach for running PHP binaries. This is a popular approach used by many web hosts. For these systems, the php process runs as the owner of the php files themselves, allowing for a simpler configuration and a more secure environment for the specific case of shared hosting.

Note: suexec methods should NEVER be used on a single-site server configuration, they are more secure only for the specific case of shared hosting.

In such an suexec configuration, the correct permissions scheme is simple to understand.

All files should be owned by the actual user's account, not the user account used for the httpd process.
Group ownership is irrelevant, unless there's specific group requirements for the web-server process permissions checking. This is not usually the case.

All directories should be 755 or 750.

All files should be 644 or 640. Exception: wp-config.php should be 440 or 400 to prevent other users on the server from reading it.

No directories should ever be given 777, even upload directories. Since the php process is running as the owner of the files, it gets the owners permissions and can write to even a 755 directory.`

**Comment by rallek** [comment]
unpacking was the issue! @cmfcmf helped me to figure out I have to use ``-p`` in the tar as well.

but still I have to manage the permissions of cache  The directory ``app/cache/prod/annotations`` (and others I guess) if I use CLI. This is because now a wrong owner owns the files. Will try web installation next

**Comment by rallek** [comment]
even if installation is done by web browser I run into troubble with permissions:

````
Oops! An Error Occurred
The server returned a "500 Internal Server Error".
Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

Failed to copy "/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/parameters.yml" to "/www/htdocs/w012c698/neu.indaus.de/Zikula_Core-1.4.0/app/config/custom_parameters.yml" because target file could not be opened for writing.````

I have to make ``app/config`` and ``parameters.yml`` 777. 755 is not enough.

Than the installer starts checking permissions. Two other directories must be also 777. At the end I could finish the installation via web now.

This looks not very professional :-(

**Comment by craigh** [comment]
> This looks not very professional :-(

please address problems and suggest solutions instead of making derogatory comments like this. you may not mean to, but you are insulting people and their work.

**Comment by rallek** [comment]
Sorry, that was not what I want.

**Comment by rallek** [comment]
Next try with 575 looks better but do have stilltrouble. I tried again CLI. This runs fine. But calling the website gives an error. I learned yesterday I have to make parameters.yml from 755 to 777. I did this. Next is the permission of app/chache/prod is not fine with 755. So I change this to 777 as well.

Next one ist: app/cache/prod/doctrine/orm/DoctrineProxy
and next: app/cache/prod/ztemp/Theme_compiled

here I stopp my CLI try. Next I will delete everything, unpack it again and try via web.

For record: All files I have seen are with the right ownership. No one owned by the webserver anymore. Let us see how this will work via web.
